cli/command/container/run.go

```diff
@@ -123,6 +128,7 @@ func runContainer(dockerCli *command.DockerCli, opts *runOptions, copts *contain
 	hostConfig := containerConfig.HostConfig
 	stdout, stderr := dockerCli.Out(), dockerCli.Err()
 	client := dockerCli.Client()
+	ociPlatform := system.ParsePlatform(opts.platform)
```

Comment:
I think this should be parsed by the flag instead after flag parsing. To accomplish this you can create a new `PlatformOpt` type in `opts/platform.go` with a `func Value() spec.Platform` to return the parsed type.

The field in the options struct will be a `PlatformOpt` instead of a `string`